zivid_camera/src/settings_generator.cpp

```diff
@@ -205,7 +267,7 @@ class DynamicReconfigureCfgGenerator
     res << "gen = ParameterGenerator()\n";
     if (insert_enabled_)
     {
-      res << "gen.add(\"enabled\", bool_t, 0, \"When frame is enabled it will be included in captures\", "
+      res << "gen.add(\"enabled\", bool_t, 0, \"When this acquisition is enabled it will be included in captures\", "
              "False)\n";
     }
     res << ss_.str();
```

Comment:
I know specifically this has not changed in the PR here, but is there any reason not to call `apply()` directly here (removing `ss_`), rather than calling apply in
https://github.com/zivid/zivid-ros/blob/b050280eddb2e44715e5b581c3fddea4d3d233d3/zivid_camera/src/settings_generator.cpp#L500
?
Same question for the header utils generator.

Reply:
I don't understand this comment fully. Apply is called from the outside for all of the settings, when we loop over all of the settings. This is why the output is stored in ss_. This str() is called later when writing the files to disk.

Reply:
So it's because we want to traverse the settings tree outside and apply them to the generator? Which then combines this in the end with the boiler-plate part of the file

Reply:
yes